backend/app/controllers/spree/admin/store_credits_controller.rb

```diff
@@ -74,7 +74,15 @@ def permitted_resource_params
           merge(created_by: try_spree_current_user)
       end
 
+      def user_exists?
+        return unless parent.nil?
+
+        flash[:error] = flash_message_for(model_class.new, :not_found)
+        redirect_to admin_users_path
```

Comment:
If we generalized this not found logic for parent classes we wouldn't need these specialized implementations that vary only in the path.